tinytuya/MappedDevice.py

```diff
@@ -0,0 +1,756 @@
+# TinyTuya Mapped Device
+# -*- coding: utf-8 -*-
+"""
+ Python module to map Tuya DPs to names
+
+ Author: uzlonewolf https://github.com/uzlonewolf
+ For more information see https://github.com/jasonacox/tinytuya
+
+ Local Control Classes
+    MappedDevice(..., product_id="...", mappingfile=DPMAPPINGSFILE, mapping=None, cloud=None)
+        product_id (str): Product ID string to match in mapping file
+        mappingfile (str, optional): Name of JSON file to load mapping data from.  Default: 
+        mapping (dict, optional): Additional mapping data to use in addition to loaded file
+        cloud (Cloud object): Initilized tinytuya.Cloud object to use to download the mapping if needed
+
+        See OutletDevice() for common constructor arguments
+
+        At least one of product_id+mappingfile, mapping, or cloud must be provided!
+        If product_id is not provied, a lookup for it in DEVICEFILE will be attempted
+
+ Functions
+    MappedDevice
+        set_mapping(mapping_dict)
+
+    Inherited
+        See OutletDevice()
+
+ Attributes
+    .dps
+        Set or Get mapped values
+            val = MappedDevice.dps.some_dp_name
+              or
+            val = MappedDevice.dps['some_dp_name']
+              or
+            MappedDevice.dps.some_dp_name = new_value
+              or
+            MappedDevice.dps['some_dp_name'] = new_value

```

Comment:
This is brilliant. It elevates ease-of-use. I'm thinking how we best add this to README...